### PR TITLE
Run the remove-linked-state-mixin codemod

### PR DESCRIPTION
**Pull request by jsnajdr**
This runs the remove-linked-state-mixin codemod that landed in #13925.

The second commit removes the `react-addons-linked-state-mixin` from `package.json` and reshrinkwraps.

**To test:**
Go through the forms in the affected files and check that they work:
```
client/auth/login.jsx
client/me/application-passwords/index.jsx
client/me/help/help-contact-form/index.jsx
client/me/profile-links-add-other/index.jsx
client/me/reauth-required/index.jsx
client/me/security-2fa-backup-codes-prompt/index.jsx
client/me/security-2fa-code-prompt/index.jsx
client/me/security-2fa-enable/index.jsx
client/me/security-account-recovery/edit-email.jsx
```

### CODE REVIEW

**Comment by jsnajdr** [comment]
This runs the `remove-linked-state-mixin` codemod that landed in #13925.

The second commit removes the `react-addons-linked-state-mixin` from `package.json` and reshrinkwraps.


**To test:**
Go through the forms in the affected files and check that they work:
```
client/auth/login.jsx
client/me/application-passwords/index.jsx
client/me/help/help-contact-form/index.jsx
client/me/profile-links-add-other/index.jsx
client/me/reauth-required/index.jsx
client/me/security-2fa-backup-codes-prompt/index.jsx
client/me/security-2fa-code-prompt/index.jsx
client/me/security-2fa-enable/index.jsx
client/me/security-account-recovery/edit-email.jsx
```

File: client/auth/login.jsx [hunk 1]
```diff
@@ -180,6 +181,11 @@ export const Login = createReactClass( {
 			</Main>
 		);
 	},
+
+	handleChange( e ) {
```
Comment (blowery) [inline client/auth/login.jsx]: my gut said "shouldn't this be bound" but then I noticed this is a `createClass` class, not an es6 class. Just leaving this here as a warning to other reviewers.

Comment (jsnajdr) [reply client/auth/login.jsx]: Yes, all components involved in this PR should be `createClass` classes. It's about removing a mixin, something that can't happen in a ES6 class.

File: client/auth/login.jsx [hunk 2]
```diff
@@ -113,7 +111,8 @@ export const Login = createReactClass( {
 									disabled={ requires2fa || inProgress }
 									placeholder={ translate( 'Username or email address' ) }
 									onFocus={ this.recordFocusEvent( 'Username or email address' ) }
-									valueLink={ this.linkState( 'login' ) }
+									value={ this.state.login }
```
Comment (blowery) [inline client/auth/login.jsx]: other note to other reviewers, yes, these usages of `state` are safe. `linked-state-mixin` required component state to work, so if this stuff worked previously, there must have been a `getInitialState` method defined. You don't really have to check every component to make sure it also has a `getInitialState` method.

**Comment by blowery** [comment]
tested out the various flows that were affected, didn't see any problems.

File: client/me/security-2fa-enable/index.jsx [hunk 3]
```diff
@@ -457,5 +455,10 @@ export default localize(
 				</div>
 			);
 		},
+
+		handleChange( e ) {
+			const { name, value } = e.currentTarget;
```
Comment (samouri) [inline client/me/security-2fa-enable/index.jsx]: i like this, I'd never seen it before for reusing an onChange handler

File: client/me/security-2fa-code-prompt/index.jsx [hunk 4]
```diff
@@ -217,13 +214,14 @@ export default localize(
 							autoFocus
 							className="security-2fa-code-prompt__verification-code"
 							disabled={ this.state.submittingForm }
-							name="verification-code"
```
Comment (samouri) [inline client/me/security-2fa-code-prompt/index.jsx]: should we be checking that no other functionality directly depended on the name being "verification-code"?

File: client/me/security-2fa-enable/index.jsx (hunk 3 above)
Comment (jsnajdr) [reply client/me/security-2fa-enable/index.jsx]: It's used at a lot of other places, for example [here](https://github.com/Automattic/wp-calypso/blob/38040768fba5ee8539ebd69204a817f644aa15f3/client/my-sites/checkout/checkout/paypal-payment-box.jsx#L52-L57). The principle is: be a know-nothing method and extract everything you need from the target element 🙂

File: client/me/security-2fa-code-prompt/index.jsx (hunk 4 above)
Comment (jsnajdr) [reply client/me/security-2fa-code-prompt/index.jsx]: Yes, that's a real risk. For example, there could be CSS rules that depend on the name: `.field[name="verification-code"]`. But I didn't find anything.

File: client/me/application-passwords/index.jsx [hunk 5]
```diff
@@ -110,9 +108,10 @@ const ApplicationPasswords = React.createClass( {
 							className="application-passwords__add-new-field"
 							disabled={ this.state.submittingForm }
 							id="application-name"
-							name="application-name"
+							name="applicationName"
```
Comment (samouri) [inline client/me/application-passwords/index.jsx]: Does this line need to change to?

`<FormLabel htmlFor="application-name">`

File: client/me/security-2fa-backup-codes-prompt/index.jsx [hunk 6]
```diff
@@ -144,17 +141,18 @@ export default localize(
 						</FormLabel>
 						<FormTelInput
 							disabled={ this.state.submittingCode }
-							name="backup-code-entry"
```
Comment (samouri) [inline client/me/security-2fa-backup-codes-prompt/index.jsx]: this one has an `htmlFor` as well:
`<FormLabel htmlFor="backup-code-entry">`

File: client/me/security-2fa-code-prompt/index.jsx (hunk 4 above)
Comment (samouri) [reply client/me/security-2fa-code-prompt/index.jsx]: same as above: `<FormLabel htmlFor="verification-code">` may need to be changed

**Comment by samouri** [comment]
I grepped around for all the "name" changes.  I don't know `redux-form` well enough to know the importance of the `htmlFor` and if it needs to be in sync with the modified names.

If not, then feel free to ship.  
If yes, then fix + let me know what it does :)

File: client/me/security-2fa-code-prompt/index.jsx (hunk 4 above)
Comment (jsnajdr) [reply client/me/security-2fa-code-prompt/index.jsx]: `htmlFor` refers to the `id` attribute, not `name`. So the value doesn't need to change. We could unify the naming for `id`s and `name`s, of course, but it's not strictly necessary.

Comment (samouri) [reply client/me/security-2fa-code-prompt/index.jsx]: Ok, then feel free to ignore all my comments on `htmlFor`. I didn't find any other usages of the names you changed when I grepped about

**Comment by samouri** [comment]
thanks for pushing this past the finish line @jsnajdr! we are one step closer now 🎉

**Comment by jsnajdr** [comment]
Thanks @samouri, I started working on this codemod in May, so it was indeed about time to ship it 😉

**Pull request merged**